py/torch_tensorrt/dynamo/conversion/impl/reduce.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional, Tuple, Union
+
+import tensorrt as trt
+from torch.fx.node import Target
+from torch_tensorrt.dynamo._SourceIR import SourceIR
+from torch_tensorrt.dynamo.conversion.converter_utils import (
+    cast_trt_tensor,
+    get_axes_for_reduce_op,
+)
+from torch_tensorrt.fx.converters.converter_utils import set_layer_name
+from torch_tensorrt.fx.types import TRTNetwork, TRTTensor
+
+
+def amax(
+    network: TRTNetwork,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_val: TRTTensor,
+    dim: Union[int, Tuple[int]],
+    keepdim: bool = False,
+) -> TRTTensor:
+    if (isinstance(input_val, TRTTensor)) and (
+        input_val.dtype == trt.int8 or input_val.dtype == trt.int32
+    ):
+        input_val = cast_trt_tensor(network, input_val, trt.float32, name)
```

Comment:
Is this required for the `IReduceLayer`?

Reply:
Yes, it's required because `TypeError: Provided an unsupported data type as an input data type (support: bool, int32, long, half, float), got: torch.int8` if I test int8. Even though the [doc](https://docs.nvidia.com/deeplearning/tensorrt/operators/docs/Reduce.html#data-types) says Reduce Layer supports int32, but I tested it doesn't support.